esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -1124,10 +1124,31 @@ bool CWsWorkunitsEx::onWUSyntaxCheckECL(IEspContext &context, IEspWUSyntaxCheckR
         Owned<IConstWorkUnit> cw(factory->openWorkUnit(wuid.str()));
         WsWUExceptions errors(*cw);
         resp.setErrors(errors);
+
+        StringBuffer msg;
+        WUState st = cw->getState();
         cw.clear();
 
-        if (!factory->deleteWorkUnit(wuid.str()))
-            throw MakeStringException(ECLWATCH_CANNOT_DELETE_WORKUNIT, "%s: Workunit cannot be deleted. Please check ESP log.", wuid.str());
+        switch (st)
+        {
+        case WUStateAborted:
+        case WUStateCompleted:
+        case WUStateFailed:
+            if (!factory->deleteWorkUnit(wuid.str()))
+                throw MakeStringException(ECLWATCH_CANNOT_DELETE_WORKUNIT, "Workunit %s cannot be deleted.", wuid.str());
+            break;
+        default:
+            secAbortWorkUnit(wuid.str(), *context.querySecManager(), *context.queryUser());
+            if (!factory->deleteWorkUnit(wuid.str()))
+            {
+                throw MakeStringException(ECLWATCH_CANNOT_DELETE_WORKUNIT,
+                    "WUSyntaxCheckECL is timed out. Workunit %s cannot be deleted now. You may delete it when its status is allowed.", wuid.str());
```

Comment:
I think it will sound better if you remove "is" or change it to "has".

Maybe "You may delete it when its status changes"?